File: Basic-Video-Chat-ConnectionService_FCM/app/src/main/java/com/tokbox/sample/basicvideochat_connectionservice/VonageConnectionHolder.java
package com.tokbox.sample.basicvideochat_connectionservice;

public class VonageConnectionHolder {

    private static VonageConnectionHolder instance;
    private VonageConnection connection;

    private VonageConnectionHolder() {}

    public static synchronized VonageConnectionHolder getInstance() {
        if (instance == null) {
            instance = new VonageConnectionHolder();
        }
        return instance;
    }

    public VonageConnection getConnection() {
        return connection;
    }

    public void setConnection(VonageConnection connection) {
        this.connection = connection;
    }
}

File: Basic-Video-Chat-ConnectionService-Kotlin/app/src/main/java/com/tokbox/sample/basicvideochatconnectionservice/connectionservice/VonageConnectionHolder.kt
package com.tokbox.sample.basicvideochatconnectionservice.connectionservice

import android.annotation.SuppressLint
import com.tokbox.basic_video_chat_connectionservice.connectionservice.VonageConnection

object VonageConnectionHolder {
    @SuppressLint("StaticFieldLeak")
    var connection: VonageConnection? = null
}